### PR TITLE
build(client-utils): use export map instead of 'browser' field

### PR DESCRIPTION
**Pull request by DLehenbauer**
Previously, we used [the 'browser' field](https://github.com/defunctzombie/package-browser-field-spec) to implement different entry points for browser vs. node.

This field does not appear to be supported by CloudPack.  Instead, we now use [conditional exports](https://nodejs.org/docs/latest-v18.x/api/packages.html#conditional-exports) in the export map.

### CODE REVIEW

File: common/lib/common-utils/package.json
Comment (tylerbutler) [inline common/lib/common-utils/package.json:1]: This whole package is deprecated... Can we skip it? Even if we update it, none of our packages use it so we have no reason to publish a new version.

File: packages/common/client-utils/package.json [hunk 1]
```diff
@@ -14,22 +14,60 @@
...
+				},
+				"require": {
+					"types": "./dist/bufferBrowser.d.ts",
+					"default": "./dist/bufferBrowser.js"
+				}
 			}
 		}
 	},
 	"main": "lib/indexNode.js",
 	"browser": {
```
Comment (tylerbutler) [inline packages/common/client-utils/package.json:62]: Can we delete this altogether?

File: common/lib/common-utils/package.json
Comment (DLehenbauer) [reply common/lib/common-utils/package.json:1]: I reviewed dependencies (assert, LazyPromise, and TypedEventEmitter) and they are all available elsewhere.

Reverting changes to common-utils.  Thanks!

File: packages/common/client-utils/src/base64Encoding.ts [hunk 2]
```diff
@@ -3,7 +3,8 @@
  * Licensed under the MIT License.
  */
 
-import { IsoBuffer } from "./indexNode.js";
+// Note: The exports map in package.json will correct this import to "./bufferNode.js" in node environments
+import { IsoBuffer } from "./bufferBrowser.js";
```
Comment (DLehenbauer) [inline packages/common/client-utils/src/base64Encoding.ts:7]: Note: changed to 'bufferBrowser.js' instead of 'indexBrowser.js' to avoid import cycle.

**Review by tylerbutler — APPROVED** [review]

File: packages/common/client-utils/package.json [hunk 3]
```diff
@@ -14,23 +14,51 @@
...
 			}
 		}
 	},
 	"main": "lib/indexNode.js",
-	"browser": {
-		"./dist/indexNode.js": "./dist/indexBrowser.js",
-		"./dist/indexNode.d.ts": "./dist/indexBrowser.d.ts",
-		"./lib/indexNode.js": "./lib/indexBrowser.js",
-		"./lib/indexNode.d.ts": "./lib/indexBrowser.d.ts"
-	},
```
Comment (jason-ha) [inline packages/common/client-utils/package.json:33]: Removing these means any uses of indexNode.js will not be redirected to indexBrowser.js. trace.ts uses indexNode.js; so, that needs corrected.

Comment (jason-ha) [inline packages/common/client-utils/package.json:33]: This is also replicated in src/cjs/package.json that is needed for webpack (with CJS). AB#7374 would help avoid getting out of sync.

Comment (tylerbutler) [reply packages/common/client-utils/package.json:33]: Related issue: #21252

**Pull request merged**